core/src/chain/header.rs

```diff
@@ -0,0 +1,108 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash as _Hash, Hasher};
+
+use chain::entry::Entry;
+use chain::chain::SourceChain;
+
+/// Properties defined in HeadersEntrySchema from golang alpha 1
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[allow(non_snake_case)]
+pub struct Header {
+    /// the type of this entry
+    /// system types may have associated "subconscious" behavior
+    Type: String,
+    /// ISO8601 time stamp
+    Time: String,
+    /// optional link to the immediately preceding header in the chain
```

Comment:
this is optional only because it could be the last entry in the chain.  It would be good to include that in a comment here just for making the reason  for optionality transparent.

Reply:
fair